distributed/client.py

```diff
@@ -358,13 +358,17 @@ def executor(self):
         return self.client
 
     @property
-    def status(self):
-        """Returns the status
+    def status(
+        self,
+    ) -> Literal["cancelled", "error", "finished", "lost", "pending"] | None:
```

Comment:
Could we define this as a custom type at the top of the file and then use it here. That way users can use the type themselves instead of having to duplicate it.